comp/remote-config/rcclient/rcclient.go

```diff
@@ -66,12 +88,53 @@ func (rc rcClient) Listen() error {
 	rc.taskProcessed = map[string]bool{}
 
 	rc.client.Subscribe(state.ProductAgentTask, rc.agentTaskUpdateCallback)
+	rc.client.Subscribe(state.ProductAgentConfig, rc.agentConfigUpdateCallback)
 
 	rc.client.Start()
 
 	return nil
 }
 
+func (rc rcClient) agentConfigUpdateCallback(updates map[string]state.RawConfig) {
+	mergedConfig, err := remote.MergeRCAgentConfig(rc.client, updates)
+	if err != nil {
+		return
+	}
+
+	// TODO RCM-1064: implement priority between CLI and remote-config
+	// If there is no error, override the configs
+	if len(mergedConfig.LogLevel) > 0 {
```

Comment:
I'm not entirely sure of how RC works here, but won't this be happening every time the Agent refreshes/receives RC config, even if there are no changes?  Meaning we would have the log message below regularly and the `SetRuntimeSetting()` call even if the log level is not changing?

Reply:
For now (and probably a few more releases) only the log_level is supported, so this will happen only if the Agent receives a new LogLevel config change, so `SetRuntimeSetting` is called pretty rarely.
I've just extended the condition so it does not call `SetRuntimeSetting` if the log level remains the same in an update